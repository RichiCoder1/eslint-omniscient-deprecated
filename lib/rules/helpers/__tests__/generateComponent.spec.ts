import {
    ClassInformation,
    generateClassComponent,
    FunctionInformation,
    generateFunctionComponent,
} from "../generateComponent";

describe("generateComponent", () => {
    it("should generate class component", () => {
        const classDefinition: ClassInformation = {
            name: "Test",
            wrapped: false,
            staticProperties: ["displayName = 'Test';"],
            renderBody: "{ return this.getA(); }",
            areEqualFunction: "is",
            extendsName: "Component",
            constructorLines: [`this.state = {a : 1};`],
            instanceProperties: [`getA = () => { return <h1>Test</h1>; };`],
        };

        const output = generateClassComponent(classDefinition, true);
        expect(output).toMatchSnapshot();
    });

    it("should generate wrapped class component", () => {
        const classDefinition: ClassInformation = {
            name: "Test",
            wrapped: true,
            staticProperties: ["displayName = 'Test';"],
            renderBody: "{ return null; }",
            areEqualFunction: "is",
            extendsName: "Component",
            constructorLines: [],
            instanceProperties: [],
        };

        const output = generateClassComponent(classDefinition, true);
        expect(output).toMatchSnapshot();
    });

    it("should generate class component without class properties", () => {
        const classDefinition: ClassInformation = {
            name: "Test",
            wrapped: false,
            staticProperties: ["displayName = 'Test';"],
            renderBody: "{ return this.getA().bind(this); }",
            areEqualFunction: "is",
            extendsName: "Component",
            constructorLines: [`this.state = {a : 1};`],
            instanceProperties: [`getA() { return <h1>Test</h1>; };`],
        };

        const output = generateClassComponent(classDefinition, false);
        expect(output).toMatchSnapshot();
    });

    it("should generate function component", () => {
        const funcDefinition: FunctionInformation = {
            name: "Test",
            wrapped: false,
            staticProperties: ["displayName = 'Test';"],
            props: "{a}",
            renderBody: "{ return a; }",
            areEqualFunction: "is",
        };

        const output = generateFunctionComponent(funcDefinition, null);
        expect(output).toMatchSnapshot();
    });

    it("should generate wrapped function component", () => {
        const funcDefinition: FunctionInformation = {
            name: "Test",
            wrapped: true,
            staticProperties: ["displayName = 'Test';"],
            props: "{a}",
            renderBody: "{ return a; }",
            areEqualFunction: "is",
        };

        const output = generateFunctionComponent(funcDefinition, null);
        expect(output).toMatchSnapshot();
    });

    it("should generate memoized function component", () => {
        const funcDefinition: FunctionInformation = {
            name: "Test",
            wrapped: false,
            staticProperties: ["displayName = 'Test';"],
            props: "{a}",
            renderBody: "{ return a; }",
            areEqualFunction: "is",
        };

        const output = generateFunctionComponent(funcDefinition, "React.memo");
        expect(output).toMatchSnapshot();
    });
});
