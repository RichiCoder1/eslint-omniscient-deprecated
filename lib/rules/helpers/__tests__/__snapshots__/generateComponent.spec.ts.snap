// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateComponent should generate class component 1`] = `
"class Test extends Component {
constructor(props) {
super(props);
this.state = {a : 1};
}

getA = () => { return <h1>Test</h1>; };

shouldComponentUpdate(nextProps, nextState) {
    return !is(this.props, nextProps) || !is(this.state, nextState);
}

render() { return this.getA(); }

static displayName = 'Test';
}
"
`;

exports[`generateComponent should generate class component without class properties 1`] = `
"class Test extends Component {
constructor(props) {
super(props);
this.state = {a : 1};
}

getA() { return <h1>Test</h1>; };

shouldComponentUpdate(nextProps, nextState) {
    return !is(this.props, nextProps) || !is(this.state, nextState);
}

render() { return this.getA().bind(this); }
}

Test.displayName = 'Test';"
`;

exports[`generateComponent should generate function component 1`] = `
"function Test({a}) { return a; }
Test.displayName = 'Test';"
`;

exports[`generateComponent should generate memoized function component 1`] = `
"const Test = React.memo(function Test({a}) { return a; }, is);
Test.displayName = 'Test';"
`;

exports[`generateComponent should generate wrapped class component 1`] = `
"(() => {
class Test extends Component {
shouldComponentUpdate(nextProps, nextState) {
    return !is(this.props, nextProps) || !is(this.state, nextState);
}

render() { return null; }

static displayName = 'Test';
}

return Test;
})()"
`;

exports[`generateComponent should generate wrapped function component 1`] = `
"(() => {
function Test({a}) { return a; }
Test.displayName = 'Test';
return Test;
})()"
`;
